### PR TITLE
fix: skip empty value for array locations using init

### PR DESCRIPTION
**Pull request by shahzad31**
fixes https://github.com/elastic/synthetics/issues/658

### CODE REVIEW

File: __tests__/generator/utils.test.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+describe('Generator utils', () => {
+  it('does not add empty values in array', async () => {
+    const template = readFileSync(
+      join(__dirname, '../../templates', 'synthetics.config.ts'),
+      'utf-8'
+    );
+    const values = {
+      locations: ['us_east'],
+      privateLocations: [],
```
Comment (vigneshshanmugam) [inline __tests__/generator/utils.test.ts:38]: shouldn't this be `['']` to actually test this fix?

File: .gitignore [hunk 2]
```diff
@@ -119,3 +119,5 @@ __tests__/e2e/junit.xml
 
 .idea/
 seccomp/build
+
+test-projects/
```
Comment (vigneshshanmugam) [inline .gitignore:123]: not sure why we need this on .gitignore?

File: __tests__/generator/utils.test.ts [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+ */
+
+import { readFileSync } from 'fs';
+import { join } from 'path';
+import { replaceTemplates } from '../../src/generator/utils';
+
+describe('Generator utils', () => {
+  it('does not add empty values in array', async () => {
+    const template = readFileSync(
+      join(__dirname, '../../templates', 'synthetics.config.ts'),
```
Comment (vigneshshanmugam) [inline __tests__/generator/utils.test.ts:33]: nit: join(dirname, "..", "..", "templates", "synthetics.config.ts")

**Review by vigneshshanmugam — APPROVED** [review]
LGTM except for the test comment.

File: .gitignore (hunk 2 above)
Comment (shahzad31) [reply .gitignore:123]: this is to init test projects if you have any as part of the repo.

**Pull request merged**